src/wasm-linker.cpp

```diff
@@ -297,114 +298,38 @@ void Linker::emscriptenGlue(std::ostream& o) {
     WasmPrinter::printModule(&out.wasm, std::cerr);
   }
 
-  out.wasm.removeImport(EMSCRIPTEN_ASM_CONST); // we create _sig versions
-
-  makeDynCallThunks();
-
-  o << ";; METADATA: { ";
-  // find asmConst calls, and emit their metadata
-  struct AsmConstWalker : public PostWalker<AsmConstWalker, Visitor<AsmConstWalker>> {
-    Linker* parent;
-
-    std::map<std::string, std::set<std::string>> sigsForCode;
-    std::map<std::string, Address> ids;
-    std::set<std::string> allSigs;
+  for (auto f : emscripten::makeDynCallThunks(out.wasm, getTableData())) {
```

Comment:
This is the one where we really need it :)
If we don't have any entries in the table, here we want to return an empty container that we can ignore. If we wanted a ref, we need to take it from the table, which is empty, so we would either assert or index-out-of-bounds.

In particular splitting this into ref and non-ref versions lets us modify the data conveniently, and read the data, without needing to guard against `size > 0` so we don't mutate it, is the intent